### PR TITLE
task/DES-2939 -  New App Notes Field "hideQueue" 

### PR DESCRIPTION
**Pull request by fnets**
## Overview: ##
Added a new field for Apps that will hide the queue selection for apps. This is intended only for use with Compress and Extract apps

## PR Status: ##

* [x] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2939](https://tacc-main.atlassian.net/browse/DES-2939)

## Summary of Changes: ##
Added a boolean that hides the app's queue and allocation if set to `true`.

## Testing Steps: ##
1. Go to designsafe.dev/rw/workspace/compress
2. Check to make sure that the Queue and Allocation selection menus are **not** available through the job submission pipeline
3. Go to any other app (other than Extract) and make sure that the Queue and Allocation selection menus **are** available

### CODE REVIEW

File: client/modules/workspace/src/AppsWizard/AppsFormSchema.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export const getConfigurationFields = (
     definition.jobAttributes.execSystemId
   ) as TTapisSystem;
 
-  if (definition.jobType === 'BATCH') {
+  if (definition.jobType === 'BATCH' && !definition.notes.hideQueue) {
```
Comment (rstijerina) [inline client/modules/workspace/src/AppsWizard/AppsFormSchema.ts:178]: do we need to params here? one for `hideQueue` and one for `hideAllocation`?

File: client/modules/workspace/src/AppsWizard/AppsFormSchema.ts [hunk 2]
```diff
@@ -472,7 +472,7 @@ const FormSchema = (
     queue_name: definition.jobAttributes.execSystemLogicalQueue,
   }) as TTapisSystemQueue;
 
-  if (definition.jobType === 'BATCH') {
+  if (definition.jobType === 'BATCH' && !definition.notes.hideQueue) {
```
Comment (rstijerina) [inline client/modules/workspace/src/AppsWizard/AppsFormSchema.ts:475]: just checking - does the job submit with defaults set even if hidden here?

Comment (fnets) [reply client/modules/workspace/src/AppsWizard/AppsFormSchema.ts]: You were right, and I addressed this in my latest commit.

**Review by rstijerina — APPROVED** [review]

**Review by sophia-massie — APPROVED** [review]
LGTM!

**Pull request merged**